source/agora/common/Amount.d

```diff
@@ -86,6 +86,14 @@ public struct Amount
         return ret;
     }
 
+    /// Non-throwing version of toString
+    /// Use it only for debugging when called from nothrow function
+    public string toStringNT () const @safe nothrow
+    {
+        try return this.toString();
+        catch (Exception e) {return "";}
```

Comment:
Maybe this return string should indicate the exception.

Reply:
Hmm, maybe return null would make sense